src/lib.rs

```diff
@@ -934,19 +930,102 @@ impl DstLayout {
     }
 }
 
-/// A trait which carries information about a type's layout that is used by the
-/// internals of this crate.
+/// Implements [`KnownLayout`].
+///
+/// This derive analyzes various qualities of a type's layout that are needed
+/// for some of zerocopy's APIs. It should be used whenver possible, and can be
```

Comment:
Should we be suggesting "use this whenever possible", or just leave it up to users to discover (via bounds) when they need to derive it? It seems redundant given that the bounds will guide them to when they need to derive `KnownLayout`. It adds extra prose that, to me, sounds a tad mystical.

Same elsewhere in this PR.

Reply:
Nixed the "use this whenever possible"; I agree it sounds a bit mystical here.

Reply:
That said, I'd like to keep it in the crate-level overview.